frontend/packages/dev-console/src/components/import/section/ResourceSection.tsx

```diff
@@ -31,14 +32,30 @@ const createHelpText = (k8sModel: K8sKind, helpText: string) => {
 const ResourceSection: React.FC<ResourceSectionProps> = ({ flags }) => {
   const { t } = useTranslation();
   const [field] = useField<Resources[]>('resourceTypesNotValid');
+  const fieldName = 'resources';
+  const { setFieldValue } = useFormikContext<FormikValues>();
   const invalidTypes = field.value || [];
+
   const knativeServiceAccess = useAccessReview({
     group: ServiceModel.apiGroup,
     resource: ServiceModel.plural,
     namespace: getActiveNamespace(),
     verb: 'create',
   });
+  const canIncludeKnative =
+    !invalidTypes.includes(Resources.KnativeService) &&
+    flags[FLAG_KNATIVE_SERVING_SERVICE] &&
+    knativeServiceAccess;
+
+  const [setResourceType] = useSetResourceType(flags, setFieldValue);
```

Comment:
can we make it more generic so that in future can be used elsewhere as well

```suggestion
  const [resourceType, setResourceType] = useResourceType();
```